packages/amplify-e2e-tests/src/environment/env.ts

```diff
@@ -147,16 +148,32 @@ export function getEnvironment(cwd: string, settings: { envName: string }): Prom
   but nexpect can't wait() on the spinner output
   See amplify-cli/src/initialize-env.js
 */
-export function pullEnvironment(cwd: string): Promise<void> {
-  return new Promise((resolve, reject) => {
-    spawn(getCLIPath(), ['env', 'pull'], { cwd, stripColors: true }).run((err: Error) => {
-      if (!err) {
-        resolve();
-      } else {
-        reject(err);
-      }
-    });
-  });
+
+const defaultPullEnvironmentSettings = {
+  noUpdateBackend: true,
+  appId: EOL,
+  envName: EOL,
+};
+
+export function pullEnvironment(
+  cwd: string,
+  settings?: Partial<typeof defaultPullEnvironmentSettings>,
+  testingWithLatestCodebase = false,
+): Promise<void> {
+  const s = { ...defaultPullEnvironmentSettings, ...settings };
```

Comment:
nit: Can we use destructuring to avoid the single letter variable:

```ts
  const { appId, envName } = { ...defaultPullEnvironmentSettings, ...settings };
```

Reply:
Yes. I will add this